starter-bots/purely-functional-scala/src/main/scala/Bot.scala

```diff
@@ -0,0 +1,218 @@
+  }
+
+  def directionFrom(from: Coord, to: Coord): Direction =
+    (from.y == to.y, from.x == to.x, from.x > to.x, from.y > to.y) match {
+      case (true,  false, true,  false) => West
+      case (true,  false, false, false) => East
+      case (false, true,  false, true)  => North
+      case (false, true,  false, false) => South
+      case (false, false, true,  true)  => NorthWest
+      case (false, false, false, true)  => NorthEast
+      case (false, false, true,  false) => SouthWest
+      case (false, false, false, false) => SouthEast
+    }
+
+  def aligns(from: Coord, to: Coord): Boolean =
+    from.x == to.x ||
+  from.y == to.y ||
+  (Math.abs(from.x - to.x) == Math.abs(from.y - to.y))
+
+  def inRange(from: Coord, to: Coord, range: Int): Boolean = {
+    val dx = from.x - to.x
+    val dy = from.y - to.y
+    Math.sqrt((dx * dx) + (dy * dy)) < range
+  }
+
+  def tryToShoot(currentWormId: Int, me: Player)(opponent: Opponent): Option[Decision] = for {
+    currentWorm   <- me.worms find (_.id == currentWormId)
+    val enemiesHit = hits(currentWorm, opponent)
+    if (!enemiesHit.isEmpty)
+  } yield Shoot(enemiesHit.head)
+
+  def collides(coord: Coord, opponent: OpponentWorm): Boolean =
+    coord.x == opponent.position.x &&
+  coord.y == opponent.position.y
+
+  import random._
+  def tryToMoveRandomly(
+    currentWormId: Int,
+    me: Player,
+    map: GameMap,
+    opponent: Opponent): ZIO[Random, Nothing, Option[Decision]] = (for {
+      x <- nextInt(3) map (_ - 1)
+      y <- nextInt(3) map (_ - 1)
+    } yield (x, y)) map { case (dx, dy) => for {
+      currentWorm     <- me.worms find (_.id == currentWormId)
+      val wormPosition = currentWorm.position
+      val x            = wormPosition.x + dx
+      val y            = wormPosition.y + dy
+      val mapLength    = map.length
+      if (!(x < 0 || y < 0 || x >= mapLength || y >= mapLength))
+      if (!(wormPosition.x == x && wormPosition.y == y))
+      result          <- translateToMoveOrDig(map(y)(x))
+    } yield result }
+
+  def translateToMoveOrDig: Cell => Option[Decision] = {
+    case (Cell(x, y, "AIR"))        => Some(Move(Coord(x, y)))
+    case (Cell(x, y, "DIRT"))       => Some(Dig(Coord(x, y)))
+    case (Cell(x, y, "DEEP_SPACE")) => None
+    case _                          => None
+  }
+
+  def makeMove(state: State): ZIO[Random, Any, Decision] = {
+    val opponent = state.opponents.headOption
+
+    (opponent match {
+      case Some(op) => tryToMoveRandomly(state.currentWormId, state.myPlayer, state.map, op)
+      case None     => ZIO.succeed(None)
+    }) map ((randomMove: Option[Decision]) =>
+    (Seq(opponent flatMap tryToShoot(state.currentWormId, state.myPlayer),
+      randomMove,
+      Some(DoNothing)) collect { case Some(x) => x }).head)
+  }
+
+  def startBot(roundNumber: Int): ZIO[Console with Random with FileReader, Any, Unit] = for {
+    round  <- getStrLn map (_.toInt)
```

Comment:
This `round` variable is not being used at the moment.

Reply:
Thanks.  Removed in next push.

Reply:
Minor correction: I should have said ignored.  It's still necessary to read the round to trigger the next move :)